### PR TITLE
#2911 Create Action for changing price and qty for a contract 

### PR DESCRIPTION
**Pull request by cristinamghita**
#2911 Create Action for changing price and qty for a contract

### CODE REVIEW

File: de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_ChangePriceQty.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import de.metas.process.Param;
+
+public class C_Flatrate_Term_ChangePriceQty extends JavaProcess
+{
+
+	public static final String PARAM_PRICEACTUAL = I_C_Flatrate_Term.COLUMNNAME_PriceActual;
+	public static final String PARAM_PLANNED_QTY_PER_UNIT = I_C_Flatrate_Term.COLUMNNAME_PlannedQtyPerUnit;
+
+
+	@Param(parameterName = PARAM_PRICEACTUAL, mandatory = false)
```
Comment (metas-ts) [inline de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_ChangePriceQty.java]: i would suggest to just inline the column names here, and not have the two constants..as far as i ca see therey aren't used anywhere outside this process class.
....or ...teo sais, it's ok to have them, but make them `private`

File: de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractChangePriceQtyRepository.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ */
+
+/**
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@Component
+public class ContractChangePriceQtyRepository
+{
+	public void changePriceIfNeeded(@NonNull final I_C_Flatrate_Term contract, final BigDecimal priceActual)
```
Comment (metas-ts) [inline de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractChangePriceQtyRepository.java]: i think this is not a repository..
a repositor would just get ythe entities from our DB for you..
we think this is a service
..aand, please use the `@Service` annotation..to make it more transparent..

File: de.metas.contracts/src/test/java/de/metas/contracts/impl/AbstractFlatrateTermTest.java [hunk 3]
```diff
@@ -325,8 +338,8 @@ protected I_C_Flatrate_Term createFlatrateTerm(@NonNull final I_C_Flatrate_Condi
 		contract.setDropShip_BPartner(getBpartner());
 		contract.setDropShip_Location(bpLocation);
 		contract.setDropShip_User(user);
-		contract.setPriceActual(BigDecimal.valueOf(2));
-		contract.setPlannedQtyPerUnit(BigDecimal.ONE);
+		contract.setPriceActual(PRICE);
```
Comment (metas-ts) [inline de.metas.contracts/src/test/java/de/metas/contracts/impl/AbstractFlatrateTermTest.java]: please rename to `QTY_TEN` and `PRICE_ONE`..it makes it more readable when you use them somewhere else

File: de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractChangePriceQtyRepository.java (hunk 2 above)
Comment (cristinamghita) [reply de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractChangePriceQtyRepository.java]: Yes. After reading, I also had that feeling. But wasn't sure. Is the first attempt.
Thx!

File: de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_ChangePriceQty.java (hunk 1 above)
Comment (cristinamghita) [reply de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_ChangePriceQty.java]: agree

File: de.metas.contracts/src/test/java/de/metas/contracts/impl/AbstractFlatrateTermTest.java (hunk 3 above)
Comment (cristinamghita) [reply de.metas.contracts/src/test/java/de/metas/contracts/impl/AbstractFlatrateTermTest.java]: agree

**Comment by cristinamghita** [comment]
Done! Thx for feedback!

**Pull request merged**